src/analyser.cpp

```diff
@@ -1197,9 +1197,13 @@ void Analyser::AnalyserImpl::updateUnitsMap(const ModelPtr &model,
         UnitsPtr units = model->units(unitsName);
 
         if (units->isBaseUnit()) {
-            auto iter = unitsMap.find(unitsName);
+            if (unitsMap.find(unitsName) == unitsMap.end()) {
+                unitsMap[unitsName] = unitsExponent;
+            } else {
+                auto iter = unitsMap.find(unitsName);
```

Comment:
Not sure we need to do the find twice here.

Reply:
Correct, I was just testing something. This will be corrected in a minute, although I will likely push it later with more upcoming changes.

Reply:
Still needed to push things, so that I can test locally on Windows/Linux. So, yes, addressed in commit 12b600a65.